static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -3,418 +3,730 @@ import { format } from "d3-format";
 import { utcParse, utcFormat } from "d3-time-format";
 import { scaleLinear, scaleOrdinal } from "d3-scale";
 import { schemePaired } from "d3-scale-chromatic";
-import { area, stack, curveMonotoneX, stackOrderReverse } from "d3-shape";
+import { area, stack, curveMonotoneX, stackOrderReverse, line } from "d3-shape";
 import { extent, bisector } from "d3-array";
 import { axisBottom, axisLeft } from "d3-axis";
-import { cullXAxis, cullYAxis } from "./helpers";
 import { isMobile } from "../../../libs/mobile";
 import debounce from "../../../libs/debounce";
 import { sortChannels } from "../../../libs/channels";
+import { arraysMerge } from "../../../libs/arrays";
 
-function showGraph(el) {
-  el.style.opacity = 1;
-}
+class Graph {
+  constructor(holderSelector, rawData, options) {
```

Comment:
Would be nice to have a short comment with jsdoc for those.
It's hard to conclude from the code what shape data should have, what options are available...